dlib/dnn/loss.h

```diff
@@ -1156,20 +1159,34 @@ namespace dlib
                         {
                             // Ignore boxes that can't be detected by the CNN.
                             loss -= options.loss_per_missed_target;
-                            truth_idxs.push_back(0);
+                            truth_idxs.push_back(-1);
                             continue;
                         }
                         const size_t idx = (k*output_tensor.nr() + p.y())*output_tensor.nc() + p.x();
+                        const auto i = truth_idx_to_truth_rects.find(idx);
+                        if (i != truth_idx_to_truth_rects.end())
+                        {
+                            // Ignore duplicate truth box in feature coordinates.
+                            std::cout << "Warning, ignoring object.  We encountered a truth rectangle located at " << x.rect;
```

Comment:
The logic here looks good to me.  Can you make this warning message more gramatical though?  Right now it will say `We encountered a truth rectangle located at [blah blah] because it maps to the exact same feature coordinates...`

Reply:
Fixed.